crates/next-api/src/project.rs

```diff
@@ -539,12 +548,14 @@ impl Project {
     }
 
     #[turbo_tasks::function]
-    fn project_fs(&self) -> Vc<DiskFileSystem> {
-        DiskFileSystem::new(
+    async fn project_fs(self: Vc<Self>) -> Result<Vc<DiskFileSystem>> {
+        let this = &*self.await?;
+        Ok(DiskFileSystem::new(
+            self.uri_scheme(),
```

Comment:
Why does the filesystem need to know about the url scheme? That sounds weird to me. The url scheme should only be used during code generation.

Reply:
This was the simplest way of accomplishing this. Otherwise we'll need to thread this through from other configuration through Turbopack. What do you think about `AssetContext`?